Add verbosity to CI skuba execution

dannysauer opened the pull request:
add verbosity argument to run_skuba

## Why is this PR needed?

This simple PR adds a verbosity parameter to the skuba execution method in testrunner with an elevated default value.

Fixes SUSE/avant-garde#547

## What does this PR do?

Increased default verbosity on skuba executions will increase the output recorded in the test cases, facilitating easier debugging when tests fail.  Passing it as an optional parameter allows specific calls to fine-tune the value as needed.

tdaines42 commented on ci/infra/testrunner/skuba/skuba.py:
```diff
@@ -187,4 +187,4 @@ def _run_skuba(self, cmd, cwd=None):
         env = {"SSH_AUTH_SOCK": os.path.join(self.conf.workspace, "ssh-agent-sock")}
 
-        return self.utils.runshellcommand(self.binpath + " "+ cmd, cwd=cwd, env=env)
+        return self.utils.runshellcommand(self.binpath + f" -v {verbosity} " + cmd, cwd=cwd, env=env)
```
maybe make this whole thing an f string? f"{self.binpath} -v {verbosity} {cmd}"

dannysauer replied:
Doh.  I was so excited about changing as little as possible... :)

dannysauer replied:
Updated

dannysauer commented:
Moved the default verbosity setting out to the config file, and added validation that the value provided is actually an integer.  Good suggestion, @tdaines42.  I also poked at the section I added it a little to make it flake8-friendly. :)

pablochacin commented on ci/infra/testrunner/skuba/skuba.py:
```diff
@@ -185,6 +185,15 @@ def _run_skuba(self, cmd, cwd=None):
+            v = int(verbosity)
+        except ValueError:
+            v = 2  # default verbosity if unspecified / error
+            logger.warn(f"verbosity '{verbosity}' is not an int; using {v}")
```
I would not swallow the exception silently. I would rise using the message you are using for the log.

dannysauer replied:
I guess the question is if this configuration error should halt execution.  Initially, I thought "no", but maybe you're right; maybe it should die if an invalid value is provided.

dannysauer replied:
I switched it to log as an error and raise a generic exception.  Should it just `sys.exit` instead, as there are a couple of places where _run_skuba is called inside a try block that (mildly annoyingly) catches all exceptions?  I feel like raising `Exception` is probably the technically better route, but am open to suggestions.

pablochacin replied:
I generally against exiting from inside a library. Exception should be a prefered way. And logging and raising an exception is generally considered an anti-pattern (we have been removing this from testrunner)

dannysauer replied:
I agree regarding log + exception (and regarding exit). However, several calls to this `catch exception` without emitting the exception text, so I did this out of concern about the cause of the error being lost.  Little is more annoying as a user than an error which doesn't say what's wrong. :)

Would it be better to leave this in and remove it in a separate PR improving the relevant exception handling, or just go ahead and improve the `catch` blocks' text in this PR?

dannysauer replied:
Switched to raise customized ValueError exception.

dannysauer commented:
> You should also add the verbosity option to the [Skuba configuration class ](https://github.com/SUSE/skuba/blob/master/ci/infra/testrunner/utils/constants.py#L81)and set the default value there

Ah, I didn't know about that one.  Done.

MaximilianMeister commented on ci/infra/testrunner/utils/constants.py:
```diff
@@ -83,6 +83,7 @@ def __init__(self):
             super().__init__()
             self.binpath = None
             self.srcpath = None
+            self.verbosity = 2
```
shouldnt this read from the skuba conf?

chentex replied:
I think this is here in case there is an outdated `var.yaml` file that doesn't have a value.

dannysauer replied:
Yeah; this is a default that gets overridden by the config value if present.

dannysauer commented:
Rebased on current master & squashed changes. :)

stefsuse commented on ci/infra/testrunner/skuba/skuba.py:
```diff
@@ -170,6 +170,21 @@ def _run_skuba(self, cmd, cwd=None):
             cwd = self.cwd
 
+        if verbosity is None:
+           verbosity = self.conf.skuba.verbosity  # noqa: E111
```
indent is off, need one more space

dannysauer replied:
I thought it looked better with the variables vertically aligned, feels messed up with that weird one-char offset. Python needs to support `iff` :)

dannysauer replied:
Extra space added so it's wonky like the preceding block. ;)

stefsuse commented on ci/infra/testrunner/skuba/skuba.py:
```diff
@@ -170,6 +170,21 @@ def _run_skuba(self, cmd, cwd=None):
+        except ValueError:
+            msg = f"verbosity '{verbosity}' is not an int"
+            logger.error(msg)
+            raise Exception(msg)
```
I think it makes a lot more sense to just default to 2 here like you had it in the first place, but I was late to the party..

dannysauer replied:
Whoops; I forgot to change that to `logger.exception()` and just inline the `msg`.

dannysauer replied:
I switched it to raise a more descriptive ValueError exception.  The vars file contains a default value of "2", and the class has a default of 2.  If the config file is old and the var isn't present, it'll use "2".  If the config file is new and the value is something other than 2, then the user changed it intentionally _away from_ 2 and we should notify them that they chose an invalid value.  It might make sense to check for an empty string and use that to suppress adding the `-v`; defaulting to the program's default verbosity level - but that seems an unlikely use case.

dannysauer commented:
@maximilianmaher 
> however i would set the default to 5

I only messed with it a little, and sort of guessed at 2.  Is 5 "too" verbose?  I guess it's configurable if someone wants less, which is sort of the whole point of this. :D  Done.

MaximilianMeister commented:
@dannysauer thanks for adapting the default, however now seems that the latest commit failed the author check, can you maybe squash the commits?

dannysauer commented:
Whoops - I committed with wrong email (used my personal laptop to make that change). :D  Squashed and fixed author.
